src/repository/rest/rickandmortyapi_test.go

```diff
@@ -0,0 +1,243 @@
+package rest
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"golang-bootcamp-2020/domain/model"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestRickAndMortyApi_FetchData_SinglePage(t *testing.T) {
+	restRepo := NewRickAndMortyApiRepository()
+
+	var mockServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{
+    "info": {
+        "count": 671,
+        "pages": 34,
+        "next": "https://rickandmortyapi.com/api/character/?page=2",
+        "prev": null
+    },
+    "results": [
+        {
+            "id": 1,
+            "name": "Rick Sanchez",
+            "status": "Alive",
+            "species": "Human",
+            "type": "",
+            "gender": "Male",
+            "origin": {
+                "name": "Earth (C-137)",
+                "url": "https://rickandmortyapi.com/api/location/1"
+            },
+            "location": {
+                "name": "Earth (Replacement Dimension)",
+                "url": "https://rickandmortyapi.com/api/location/20"
+            },
+            "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
+            "episode": [
+                "https://rickandmortyapi.com/api/episode/1",
+                "https://rickandmortyapi.com/api/episode/2"
+            ],
+            "url": "https://rickandmortyapi.com/api/character/1",
+            "created": "2017-11-04T18:48:46.250Z"
+        },
+        {
+            "id": 2,
+            "name": "Morty Smith",
+            "status": "Alive",
+            "species": "Human",
+            "type": "",
+            "gender": "Male",
+            "origin": {
+                "name": "Earth (C-137)",
+                "url": "https://rickandmortyapi.com/api/location/1"
+            },
+            "location": {
+                "name": "Earth (Replacement Dimension)",
+                "url": "https://rickandmortyapi.com/api/location/20"
+            },
+            "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
+            "episode": [
+                "https://rickandmortyapi.com/api/episode/1",
+                "https://rickandmortyapi.com/api/episode/2"
+            ],
+            "url": "https://rickandmortyapi.com/api/character/2",
+            "created": "2017-11-04T18:50:21.651Z"
+        }
+    ]}`))
+	}))
+	defer mockServer.Close()
+
+	expectedCharacters := []model.Character{
+		{
+			Id:       1,
+			Name:     "Rick Sanchez",
+			Status:   "Alive",
+			Species:  "Human",
+			Type:     "",
+			Gender:   "Male",
+			Origin:   model.Nested{Name: "Earth (C-137)", Url: "https://rickandmortyapi.com/api/location/1"},
```

Comment:
By convention, if you are using a hardcoded value more than 2 times, declare it as a constant